core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_plan_instruction/kurtosis_plan_instruction.go

```diff
@@ -92,13 +110,20 @@ func (builtin *KurtosisPlanInstructionWrapper) CreateBuiltin() func(thread *star
 				logrus.Debugf("Marking the plan as invalid as instruction '%s' differs from '%s'",
 					instructionWrapper.String(), scheduledInstructionPulledFromMaskMaybe.GetInstruction().String())
 			}
+			kurtosisInstructionStr := scheduledInstruction.GetInstruction().String()
+			enclaveCapabilities := scheduledInstruction.GetInstruction().GetCapabilites().GetEnclavePlanCapabilities()
+			enclavePlanInstruction := enclave_plan_instruction.NewEnclavePlanInstructionImpl(kurtosisInstructionStr, enclaveCapabilities)
+			if err := builtin.enclavePlanInstructionRepository.Save(scheduledInstruction.GetUuid(), enclavePlanInstruction); err != nil {
```

Comment:
So, it's interesting here that we now need to do both `AddInstruction` and `Save`. I'm wondering if there could be a better hierarchy here such that we calling `AddInstruction` could do both. I don't think we'd ever want to `AddInstruction` withou saving it, right?